### DISCUSSION TITLE
Can I build SSG on a specific day of the week in Vercel ?

### DISCUSSION THREAD

**Discussion by hyjoong** [vercel/next.js#49069] [Help] [answered]
### Summary

```javascript
export const getStaticProps = async () => {
  const data = await getData()
  return {
    props: {
     data
    },
  // revalidate: 604800.
  };
};
```

For example, can you rebuild every 7 p.m. on Monday??

In my opinion, even if you set the revalidate to 1 week, if it is built in the meantime, the build time will be reset again, so the revalidate will not be available.

A build can be built several times a day and once every few days, but I want to build it at a certain time of the day. Because the data is updated on a specific day of the week, at a specific time

**Comment by LimSB-dev** [comment, marked as answer]
# [How to Setup Cron Jobs on Vercel](https://vercel.com/guides/how-to-setup-cron-jobs-on-vercel#create-or-update-your-vercel.json-file)
Cron jobs are time-based scheduling tools used to automate repetitive tasks. By using a specific syntax called a [cron expression](https://vercel.com/docs/cron-jobs#cron-expressions), you can define the frequency and timing of each task. This helps improve efficiency and ensures that important processes are performed consistently.

Vercel supports cron jobs for [Serverless](https://vercel.com/docs/concepts/functions/serverless-functions) and [Edge](https://vercel.com/docs/concepts/functions/edge-functions) Functions. This guide will show you how to create your first cron job. Alternatively, you can [start from a template](https://vercel.com/templates/next.js/vercel-cron).

## [Create or update your vercel.json file](https://vercel.com/guides/how-to-setup-cron-jobs-on-vercel#create-or-update-your-vercel.json-file)
Go to your vercel.json file and add the following code:
```json
{
  "crons": [
    {
      "path": "/api/cron",
      "schedule": "0 5 * * *"
    }
  ]
}
```
The path must start with `/`

The schedule property must be a string that represents a [cron expression](https://vercel.com/docs/cron-jobs#cron-expressions). In this example, the job is scheduled to execute every day at 5:00 am UTC

```bash
# ┌───────────── minute (0 - 59)
# │ ┌───────────── hour (0 - 23)
# │ │ ┌───────────── day of the month (1 - 31)
# │ │ │ ┌───────────── month (1 - 12)
# │ │ │ │ ┌───────────── day of the week (0 - 6) (0 is Sunday, 6 is Saturday)
# │ │ │ │ │
# │ │ │ │ │
# │ │ │ │ │
# * * * * *
```

**Reply by jschuur** [reply]
Should you be able to call res.revalidate() directly from this cron invoked function, or do you need to make another fetch request from it to an API route that does that?

I just ran into a problem where my cron function didn't properly revalidate my page. See [this thread](https://github.com/vercel/next.js/discussions/42290)

**Reply by hyjoong** [reply]
I resolved it by implementing `on-demand revalidation` and `Cron Jobs` methods. Thank you for providing the information.
